File: original_php_project/src/Billing/Repository/CheckoutRepository.php
<?php

namespace App\Billing\Repository;

use App\Billing\Entity\Checkout;
use Doctrine\ORM\EntityManagerInterface;
use Doctrine\ORM\EntityRepository;

class CheckoutRepository
{
    private EntityRepository $repository;

    public function __construct(
        private EntityManagerInterface $em
    ) {
        $this->repository = $em->getRepository(Checkout::class);
    }

    public function findBySlug(string $slug): ?Checkout
    {
        return $this->repository->findOneBy([
            'slug' => $slug
        ]);
    }

    public function findById(string $id): ?Checkout
    {
        return $this->repository->findOneBy([
            'id' => $id
        ]);
    }
}

File: target_java_project/src/main/java/com/goodhelp/billing/domain/repository/CheckoutRepository.java
package com.goodhelp.billing.domain.repository;

import com.goodhelp.billing.domain.model.Checkout;

import java.util.Optional;

/**
 * Repository for checkout sessions.
 */
public interface CheckoutRepository {

    Checkout save(Checkout checkout);

    Optional<Checkout> findBySlug(String slug);
}

